spec/requests/api/vms_spec.rb

```diff
@@ -1176,6 +1176,161 @@ def update_raw_power_state(state, *vms)
     end
   end
 
+  context 'Vm assign_tags action' do
+    let(:tag1)         { {:category => "department", :name => "finance", :path => "/managed/department/finance"} }
+    let(:tag2)         { {:category => "cc",         :name => "001",     :path => "/managed/cc/001"} }
+    let(:bad_tag)      { {:category => "cc", :name => "002"} }
+
+    before do
+      FactoryGirl.create(:classification_department_with_tags)
+      FactoryGirl.create(:classification_cost_center_with_tags)
+      Classification.classify(vm2, tag1[:category], tag1[:name])
+      Classification.classify(vm2, tag2[:category], tag2[:name])
+    end
+
+    it 'can bulk assign tags to multiple VMs' do
+      api_basic_authorize collection_action_identifier(:vms, :assign_tags)
+      request_body = {
+        'action'    => 'assign_tags',
+        'resources' => [
+          { 'id' => vm1.id, 'tags' => [{ 'category' => tag1[:category], 'name' => tag1[:name] }] },
+          { 'id' => vm2.id, 'tags' => [{ 'category' => tag2[:category], 'name' => tag2[:name] }] }
+        ]
+      }
+
+      run_post(vms_url, request_body)
+
+      expected = {
+        'results' => [
+          { vm1.id.to_s => [a_hash_including('success'      => true,
+                                             'tag_category' => tag1[:category],
+                                             'tag_name'     => tag1[:name])]},
+          { vm2.id.to_s => [a_hash_including('success'      => true,
+                                             'tag_category' => tag2[:category],
+                                             'tag_name'     => tag2[:name])] }
+        ]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
```

Comment:
I would prefer action results similar to when assigning/unassigning tags on a
resource i.e. vms/72/tags, i.e. even though we can target multiple resources,
it's still an action performed (thus result) per tag.

so instead of: 

```
{
  "results": [
    {   
      "72": [
        {   
          "success": true,
          "message": "Assigning Tag: category:'department' name:'finance'",
          "href": "http://localhost:3000/api/vms/",
          "tag_category": "department",
          "tag_name": "finance"
        },  
        {   
          "success": true,
          "message": "Assigning Tag: category:'location' name:'ny'",
          "href": "http://localhost:3000/api/vms/",
          "tag_category": "location",
          "tag_name": "ny"
        }   
      ],  
      "75": [
        {   
          "success": true,
          "message": "Assigning Tag: category:'department' name:'finance'",
          "href": "http://localhost:3000/api/vms/",
          "tag_category": "department",
          "tag_name": "finance"
        }   
      ]   
    }   
  ]
}
```

I'd rather get:

```
{
  "results": [
    {   
      "success": true,
      "message": "Assigning Tag: category:'department' name:'finance'",
      "href": "http://localhost:3000/api/vms/72",
      "tag_category": "department",
      "tag_name": "finance"
    },
    {
      "success": true,
      "message": "Assigning Tag: category:'location' name:'ny'",
      "href": "http://localhost:3000/api/vms/72",
      "tag_category": "location",
      "tag_name": "ny"
    },
    {
      "success": true,
      "message": "Assigning Tag: category:'department' name:'finance'",
      "href": "http://localhost:3000/api/vms/75",
      "tag_category": "department",
      "tag_name": "finance"
    }
  ]
}
```